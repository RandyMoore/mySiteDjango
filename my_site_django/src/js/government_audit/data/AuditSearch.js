import Immutable from 'immutable';
import AuditDocument from './AuditDocument';
import NamedEntity from './NamedEntity';

function getWsScheme() {
    return window.location.protocol == "https:" ? "wss" : "ws";
}

function getNamedEntitySearchSocket() {
  if (typeof(WebSocket) !== "undefined") {
    return new WebSocket(getWsScheme() + "://" + window.location.host + "/ws/namedEntitySearch")
  }
};

const AuditSearch = Immutable.Record({
  query: '',
  queryParser: 'plain',
  years: Immutable.Set(['2017']),
  fetching: false,
  namedEntitySocket: getNamedEntitySearchSocket(),
  namedEntity: NamedEntity(),
  auditDocument: AuditDocument(),
});

export default AuditSearch;
