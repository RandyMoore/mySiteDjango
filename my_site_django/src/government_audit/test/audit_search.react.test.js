import React from 'react';
import AuditSearch from '../audit_search.react.js';
import renderer from 'react-test-renderer';

test('Default render', () => {
  const component = renderer.create(
    <AuditSearch />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render with pagination visible', () => {
  const component = renderer.create(
    <AuditSearch
      query='Ooglie booglie'
      pageCount={3}
      resultsOffset={10}
     />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
