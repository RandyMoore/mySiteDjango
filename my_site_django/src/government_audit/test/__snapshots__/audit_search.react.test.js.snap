// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Search Query 
      <br />
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <div>
    <a
      href="https://www.postgresql.org/docs/9.6/static/textsearch-controls.html"
    >
       Query Parser: 
    </a>
    <select
      onChange={[Function]}
      value="plain"
    >
      <option
        value="plain"
      >
        plain
      </option>
      <option
        value="phrase"
      >
        phrase
      </option>
      <option
        value="raw"
      >
        tsquery raw
      </option>
    </select>
  </div>
  <table
    className="results-table"
  >
    <tr>
      <th />
      <th
        width="80%"
      >
        Title
      </th>
      <th>
        Publication Date
      </th>
    </tr>
  </table>
</div>
`;

exports[`Render with pagination visible 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Search Query 
      <br />
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <div>
    <a
      href="https://www.postgresql.org/docs/9.6/static/textsearch-controls.html"
    >
       Query Parser: 
    </a>
    <select
      onChange={[Function]}
      value="plain"
    >
      <option
        value="plain"
      >
        plain
      </option>
      <option
        value="phrase"
      >
        phrase
      </option>
      <option
        value="raw"
      >
        tsquery raw
      </option>
    </select>
  </div>
  <table
    className="results-table"
  >
    <tr>
      <th />
      <th
        width="80%"
      >
        Title
      </th>
      <th>
        Publication Date
      </th>
    </tr>
  </table>
</div>
`;
